src/Sylius/Bundle/ApiBundle/Provider/LiipProductImageFilterProvider.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Provider;
+
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/** @experimental */
+class LiipProductImageFilterProvider implements ProductImageFilterProviderInterface
+{
+    use ContainerAwareTrait;
+
+    public function provideAllFilters(): ?array
+    {
+        /** @var array|null $filters */
+        $filters = $this->container->getParameter('liip_imagine.filter_sets');
+
+        return $filters;
+    }
+
+    public function provideShopFilters(): ?array
+    {
+        $filters = $this->provideAllFilters();
+
+        return $this->removeAdminFilters($filters);
+    }
+
+    private function removeAdminFilters(array $filters): array {
```

Comment:
This will be fatal error cause you pass `null` where type-hint says it must be `array`.